app/Controllers/Http/ProfilesController.ts

```diff
@@ -0,0 +1,187 @@
+
+        // Extract User from Auth
+        const user = auth.user ? auth.user : null;
+
+        // Check if user exists
+        if (user) {
+
+            // Fetch user profile
+            const userProfile = await User
+                .query()
+                .where('id', user.id)
+                .select('id', 'email')
+                .preload('profile', (profileQuery) => {
+                    profileQuery.select('name', 'mobile', 'gender', 'dob')
+                })
+                .first()
+
+
+            // Send Response to Client
+            return { message: 'User Profile', token: userProfile }
+        }
+    }
+
+
+    // ------------ Create User Profile --------------- //
+    public async create({ request, auth }: HttpContextContract) {
+
+        // Extract User from Auth
+        const user = auth.user ? auth.user : null;
+
+        // Check if user exists
+        if (user) {
+
+            const existingProfile = await Profile.findBy('userId', user.id);
+
+            // Check if user profile already exists
+            if (existingProfile) {
+                return { message: 'Profile already exists, please try updating it' }
+            }
+
+            // Schema for request body validation
+            const createUserProfileSchema = await schema.create({
+                name: schema.string({}, [
+                    rules.required(),
+                    rules.alpha({
+                        allow: ['space']
+                    }),
+                    rules.trim()
+                ]),
+                mobile: schema.string([
+                    rules.regex(/^\d{10}$/),
+                    rules.trim()
+                ]),
+                gender: schema.enum(['MALE', 'FEMALE']),
+                dob: schema.date({ format: 'yyyy-MM-dd' }, [
+                    rules.before('today')
+                ])
+            })
+
+            // Validate request body
+            const data = await request.validate({ schema: createUserProfileSchema })
+
+            Object.assign(data, { userId: user.id })
+
+            // Create user once the request body is validated
+            const profile = await Profile.create(data)
+
+            /**
+             * Refreshing the `profile` model before returning it as the payload of the response is important.
+             * If not refreshed, only the actual/few properties which were inserted will be returned
+             */
+            await profile.refresh()
+
+            // Send newly created profile to client
+            return {
+                message: 'Profile created successfully',
+                data: profile
+            }
+        }
+    }
+
+
+    // --------- Update User Profile --------------- //
+    public async update({ request, auth }: HttpContextContract) {
+
+        // Extract User from Auth
+        const user = auth.user ? auth.user : null;
+
+        // Check if user exists
+        if (user) {
```

Comment:
Do we need this if block since the auth middleware will ensure that `auth.user` will be present?

Same for all of the above methods.